### DISCUSSION TITLE
Is Presidio Analyzer Thread Safe?

### DISCUSSION THREAD

**Discussion by AlanSwansonTR** [microsoft/presidio#1213] [General]
Greetings.
My team would like to use Presidio in a multi-threaded process in order to help us increase throughput with parallelization. We have looked around to try and see if there was any documentation or topics on thread safety but had no luck.
Can you please let me know if the Analyzer is thread safe?
Thank you

**Comment by omri374** [comment]
Short answer: Yes. Long answer: Depends on the recognizers and NLP capabilities you use. Presidio relies on spaCy, transformers, stanza for named entity recognition. Last time I checked, spaCy was thread safe, but it's worth validating this.

**Reply by omri374** [reply]
We already have some support for running Presidio in batch model, leveraging spacy's batch mode: https://spacy.io/api/language#pipe

See more here: https://microsoft.github.io/presidio/samples/python/batch_processing/
